torch/csrc/jit/script/compiler.cpp

```diff
@@ -1396,17 +1397,69 @@ struct to_ir {
       const Ident& target,
       const List<Expr>& args,
       const List<Stmt>& body) {
-    // TODO: start, stop, step loop
-    if (args.size() != 1) {
+    Value *end_val, *start_val, *step_val;
+    bool isSimpleRange = (args.size() == 1);
+    if (isSimpleRange) {
```

Comment:
This is weirdly verbose, why not loop over args and ensureInt, storing into a list and check the right size.

Reply:
The thing that makes it a bit annoying is that the position of `end_val` shifts from the first position to the middle position.

I suppose I can merge `args.size() ==2 && args.size()==3` together. 

I guess putting `ensureInt` and args into a loop also shortens the code.